File: LSP+ISP/Spoilers/MarsRoverKata/Commands/ForwardCommand.cs
namespace MarsRoverKata.Commands
{
    public class ForwardCommand : ICommand
    {
        private IMovable Movable { get; set; }

        public ForwardCommand(IMovable movable)
        {
            Movable = movable;
        }

        public bool Execute()
        {
            return Movable.MoveForward();
        }
    }
}

File: DIP/Java_Final/MarsRoverKata/src/main/java/MarsRoverKata/Commands/ForwardCommand.java
package MarsRoverKata.Commands;

import MarsRoverKata.IMovable;

public class ForwardCommand implements ICommand {
    private IMovable movable;

    public IMovable getMovable() {
        return movable;
    }

    public void setRover(IMovable movable) {
        this.movable = movable;
    }

    public ForwardCommand(IMovable movable) {
        this.movable = movable;
    }

    public boolean execute() {
        return movable.moveForward();
    }
}

